### PR TITLE
Feature/expect async

### PR DESCRIPTION
**Pull request by jspringer2020**
Updated `expect` so that exceptions are handled within asynchronous processes (ie: `async` / `await`) and the `AggregateException` is examined for the expected exception type. The original discussion can be found [here](https://github.com/mattflo/NSpec/issues/83#issuecomment-125224912). Also removed some trailing whitespace.

### CODE REVIEW

**Comment by amirrajan** [comment]
I'll merge this in this weekend :+1:

**Comment by amirrajan** [comment]
@BrainCrumbz can you take a look if you have time? Give any feedback while the `async` stuff is fresh on your mind?

**Comment by BrainCrumbz** [comment]
:+1:

**Comment by jspringer2020** [comment]
I'll take a look this weekend at the tabs / spaces issue and the `closureType` and get those fixed up.

File: NSpecSpecs/describe_RunningSpecs/Exceptions/describe_expected_exception_in_act.cs [hunk 1]
```diff
@@ -53,7 +53,89 @@ void method_level_context()
 
                 context["when exception thrown from act"] = () =>
                 {
-                    actAsync = async () => await Task.Run(() => { throw new InvalidOperationException("Testing"); });
```
Comment (BrainCrumbz) [inline NSpecSpecs/describe_RunningSpecs/Exceptions/describe_expected_exception_in_act.cs]: Wouldn't it be better to leave `async`/`await`, as with all other async test cases?

**Comment by BrainCrumbz** [comment]
Besides the comments from Oct 30th and today, it looks good to me. 
Thanks for your contribution @jspringer2020 !

**Pull request merged**